exchanges/kraken/kraken_types.go

```diff
@@ -731,6 +732,84 @@ var (
 	KrakenRequestParamsTimeIOC = RequestParamsTimeForceType("IOC")
 )
 
+// FuturesPublicTrades returns public trade data
+// a terrible type for parsing data
+type FuturesPublicTrades struct {
+	ContinuationToken string `json:"continuationToken"`
+	Elements          []struct {
+		ExecutionEvent struct {
+			OuterExecutionHolder struct {
+				Execution struct {
+					LimitFilled bool `json:"limitFilled"`
+					MakerOrder  struct {
+						AccountUID          string  `json:"accountUid"`
+						ClientID            string  `json:"clientId"`
+						Direction           string  `json:"direction"`
+						Filled              string  `json:"filled"`
+						LastUpdateTimestamp int64   `json:"lastUpdateTimestamp"`
+						LimitPrice          float64 `json:"limitPrice,string"`
+						OrderType           string  `json:"orderType"`
+						Quantity            float64 `json:"quantity,string"`
+						ReduceOnly          bool    `json:"reduceOnly"`
+						SpotData            string  `json:"spotData"`
+						Timestamp           int64   `json:"timestamp"`
+						Tradeable           string  `json:"tradeable"`
+						UID                 string  `json:"uid"`
+					} `json:"makerOrder"`
+					MakerOrderData struct {
+						Fee          string `json:"fee"`
+						PositionSize string `json:"positionSize"`
```

Comment:
floats? There are a few more like MarkPrice and quantity that might need to be changed.